connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -356,14 +433,56 @@ private TaskStatus parseTaskStatus(ConnectorTaskId taskId, byte[] data) {
         }
     }
 
+    private TopicStatus parseTopicStatus(byte[] data) {
+        try {
+            SchemaAndValue schemaAndValue = converter.toConnectData(statusTopic, data);
+            if (!(schemaAndValue.value() instanceof Map)) {
+                log.error("Invalid topic status value {}", schemaAndValue.value());
+                return null;
+            }
+            @SuppressWarnings("unchecked")
+            Object innerValue = ((Map<String, Object>) schemaAndValue.value()).get(TOPIC_STATE_KEY);
+            if (!(innerValue instanceof Map)) {
+                log.error("Invalid topic status value {} for field {}", innerValue, TOPIC_STATE_KEY);
+                return null;
+            }
+            @SuppressWarnings("unchecked")
+            Map<String, Object> topicStatusMetadata = (Map<String, Object>) innerValue;
+            return new TopicStatus((String) topicStatusMetadata.get(TOPIC_NAME_KEY),
+                    (String) topicStatusMetadata.get(TOPIC_CONNECTOR_KEY),
+                    ((Long) topicStatusMetadata.get(TOPIC_TASK_KEY)).intValue(),
```

Comment:
Should we be a bit more fault tolerant here in case the active record is missing the task key -- or any other fields that are required? I understand that the current code will never write such a record, but it seems easy to do just in case users need/want to manually write active topic records with missing fields.

Reply:
That works. Refactoring existing code is probably best in a simple but separate PR.